indra/explanation/pathfinding/pathfinding.py

```diff
@@ -212,10 +219,23 @@ def shortest_path_func(G, source, target, weight, ignore_nodes,
 
 # Implementation inspired by networkx's
 # networkx.algorithms.traversal.breadth_first_search::generic_bfs_edges
-def bfs_search(g, source_node, reverse=False, depth_limit=2, path_limit=None,
-               max_per_node=5, node_filter=None, node_blacklist=None,
-               terminal_ns=None, sign=None, max_memory=int(2**29), hashes=None,
-               allow_edge=None, strict_mesh_id_filtering=False, **kwargs):
+def bfs_search(g: nx.DiGraph,
+               source_node: Node,
+               reverse: bool = False,
+               depth_limit: int = 2,
+               path_limit: Optional[int] = None,
+               max_per_node: int = 5,
+               node_filter: Optional[List[str]] = None,
+               node_blacklist: Optional[Set[Node]] = None,
+               terminal_ns: Optional[List[str]] = None,
+               sign: Optional[int] = None,
+               max_memory: int = int(2**29),
+               hashes: Optional[List[int]] = None,
+               allow_edge: Optional[Callable[[Node, Node], bool]] = None,
+               strict_mesh_id_filtering: bool = False,
+               edge_filter: Optional[Callable[[nx.DiGraph, Node, Node],
+                                              bool]] = None,
+               **kwargs) -> Generator[Tuple[Node], SendType, None]:
```

Comment:
Is this really implemented as a generator (that both accepts and sends data from `yield`?) I think you probably want `Iterable` here

Reply:
Yes, it receives send values, e.g. here: https://github.com/sorgerlab/indra/blob/master/indra/explanation/pathfinding/pathfinding.py#L336

Reply:
Amazing!!!! I have literally never seen this feature used before by anyone except David Beazley. Super cool.

Reply:
It is modeled after the way we do it on `shortest_simple_paths`